### ISSUE TITLE
Cannot exclude dependencies in JvmTestSuite declaration using Kotlin DSL

### ISSUE THREAD

**Issue by anuraaga** [gradle/gradle#19673]
### Expected Behavior

Be able to exclude dependencies for JvmTestSuite definition, or more generally, be able to define `dependencies` in the same way as `project.dependencies`

```kotlin
testing {
  suites {
    val fooTest by registering(JvmTestSuite::class) {
      dependencies {
        implementation("group:artifact:version") {
          exclude("badgroup", "badmodule")
        }
      }
    }
  }
}
```

### Current Behavior

Code cannot compile because `JvmComponentDependencies` accepts `Action<? super Dependency>` which doesn't expose `exclude`

https://docs.gradle.org/7.4-rc-1/javadoc/org/gradle/api/plugins/jvm/JvmComponentDependencies.html

### Context

When migrating from `test-sets` to test suites, I am moving declarations from `project.dependencies` to `JvmTestSuite.dependencies`, but I cannot simply move. A workaround is `with(this as ModuleDependency)` but this is unintuitive.

### Steps to Reproduce 

Above snippet is self explanatory I think

### Your Environment

Build scan URL:

**Comment by robd** [comment]
Believe this is fixed by 7.6.1 - not sure which release fixed it

**Comment by octylFractal** [comment]
The strongly typed dependency block added in #20868 fixed this issue.
